src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -866,31 +865,16 @@ private static void OnDocumentOpened(object sender, bool isCurrentContext)
 
             if (project.PushingChangesToWorkspace)
             {
-                project.ProjectTracker.NotifyWorkspace(workspace => workspace.OnDocumentOpened(document.Id, document.GetOpenTextBuffer().AsTextContainer(), isCurrentContext));
+                project.ProjectTracker.NotifyWorkspace(workspace =>
```

Comment:
Consider extracting something like `VisualStudioProjectTracker.NotifyWorkspaceOfOpenedDocument` that does this, instead of adding it in 3 places.

Reply:
I thought about it, but then I'm spending time refactoring code I'm already deleting in a branch. :smile: